extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -89,8 +88,14 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
         config.authentication.scopes.ifPresent(scopes::addAll);
 
         params.put("scopes", new JsonArray(scopes));
-        params.put("redirect_uri", buildRedirectUri(context));
-        params.put("state", generateState(context));
+
+        URI absoluteUri = URI.create(context.request().absoluteURI());
+        String dynamicPath = getDynamicPath(context, absoluteUri);
+        params.put("redirect_uri", buildRedirectUri(context, absoluteUri, dynamicPath));
+
+        if (dynamicPath != null) {
+            params.put("state", generateState(context, dynamicPath));
```

Comment:
I think we should always send the state parameter. It provides a barrier to CSRF.

Reply:
In fact, a future improvement here would be to support PKCE. Just a side comment :)

Reply:
@pedroigor You are right, I'll just keep a check that compares it with the the cookie on the way back

Reply:
PKCE is aimed at public clients. Confidential clients should use state and nonce though.

Reply:
My point is you can still have `web-app` type applications as public (although much more secure would be if people just configure their clients as confidential).

Reply:
State cookie is always used now